cmd/gitops-pusher/gitops-pusher.go

```diff
@@ -20,79 +20,142 @@ import (
 	"strings"
 	"time"
 
+	"github.com/peterbourgon/ff/v3/ffcli"
 	"github.com/tailscale/hujson"
 )
 
 var (
-	policyFname  = flag.String("policy-file", "./policy.hujson", "filename for policy file")
-	timeout      = flag.Duration("timeout", 5*time.Minute, "timeout for the entire CI run")
-	githubSyntax = flag.Bool("github-syntax", true, "use GitHub Action error syntax (https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message)")
+	rootFlagSet  = flag.NewFlagSet("gitops-pusher", flag.ExitOnError)
+	policyFname  = rootFlagSet.String("policy-file", "./policy.hujson", "filename for policy file")
+	timeout      = rootFlagSet.Duration("timeout", 5*time.Minute, "timeout for the entire CI run")
+	githubSyntax = rootFlagSet.Bool("github-syntax", true, "use GitHub Action error syntax (https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-error-message)")
 )
 
-func main() {
-	flag.Parse()
-	ctx, cancel := context.WithTimeout(context.Background(), *timeout)
-	defer cancel()
-
-	tailnet, ok := os.LookupEnv("TS_TAILNET")
-	if !ok {
-		log.Fatal("set envvar TS_TAILNET to your tailnet's name")
-	}
-	apiKey, ok := os.LookupEnv("TS_API_KEY")
-	if !ok {
-		log.Fatal("set envvar TS_API_KEY to your Tailscale API key")
-	}
-
-	switch flag.Arg(0) {
-	case "apply":
+func apply(tailnet, apiKey string) func(context.Context, []string) error {
+	return func(ctx context.Context, args []string) error {
 		controlEtag, err := getACLETag(ctx, tailnet, apiKey)
 		if err != nil {
-			log.Fatal(err)
+			return err
 		}
 
 		localEtag, err := sumFile(*policyFname)
 		if err != nil {
-			log.Fatal(err)
+			return err
 		}
 
 		log.Printf("control: %s", controlEtag)
 		log.Printf("local:   %s", localEtag)
 
 		if controlEtag == localEtag {
 			log.Println("no update needed, doing nothing")
-			os.Exit(0)
+			return nil
 		}
 
 		if err := applyNewACL(ctx, tailnet, apiKey, *policyFname, controlEtag); err != nil {
-			fmt.Println(err)
-			os.Exit(1)
+			return err
 		}
 
-	case "test":
+		return nil
+	}
+}
+
+func test(tailnet, apiKey string) func(context.Context, []string) error {
+	return func(ctx context.Context, args []string) error {
```

Comment:
am I understing this correctly? 
```ffcli.Command.Exec()``` required a function of the form ```func (context.Context, []string)``` but args are not actually be used by any of these functions?